pkg/infra/log/log.go

```diff
@@ -99,24 +106,13 @@ func (ml MultiLoggers) Log(keyvals ...interface{}) error {
 	return nil
 }
 
-// we need to implement new function for multiloggers
+// New creates a new logger from the existing one with additional context
 func (ml MultiLoggers) New(ctx ...interface{}) MultiLoggers {
-	var newloger MultiLoggers
-	for _, logWithFilter := range ml.loggers {
-		logWithFilter.val = gokitlog.With(logWithFilter.val, ctx)
```

Comment:
I might miss something, but log with filter suppose to change the log level of specific packages according to custom.ini like sqlstore debug, then it would overwrite the global configuration. by removing this are we fulfill the same functionality?

Reply:
This particular factory is used everywhere to add additional context the current logger. As in this PR it is `log.With(logger, more_context...)`. The `ml` in this case already has `logger` filter set up and this function basically wrapped it with another (random!) filter because the first element of the `ctx` is not `logger`.

Reply:
logger filter is read from settings and when we do log.New("packagename"), we check whether packagename in the filter and create a new logger with the correct level for example info. How the filter is already setup before this? Did I missed something here ```level.NewFilter(logWithFilter.val, v)``` this code basically means create me a logger with packagename and log level info (or other log levels)

Reply:
https://github.com/grafana/grafana/blob/main/conf/defaults.ini#L638 basically for this line

Reply:
Correct. I did not change the New function declared in log package. I changed the method New for struct Multilogger. For example,
```go
logger := log.New("packagename") // this logger will be set with filters because we call function New of the package log.
logWithCtx := logger.New("some", "new", "context", "data") //this is method New.
```
The method New uses the logger that was created by the function New, and therefore has the filter set. Method `New` does not declare new context `logger` and therefore, there is no need in a new filter.